data/data/agent/files/usr/local/bin/set-hostname.sh

```diff
@@ -35,7 +36,29 @@ do
         echo "Host has matching MAC address: ${filename}" 1>&2
         echo "Setting hostname to ${HOSTNAME}" 1>&2
         hostnamectl set-hostname "${HOSTNAME}"
+	hostname_set=1
     else
         echo "MAC address, ${filename}, does not exist on this host" 1>&2
     fi
 done
+
+if [[ $hostname_set -eq 0 ]]; then
```

Comment:
Just for the sake of clarity, it could be useful to move this piece into a separate function,  and add some comments about the workaround